src/inc/CrstTypes.def

```diff
@@ -745,6 +745,10 @@ Crst WrapperTemplate
     AcquiredBefore IbcProfile
 End
 
+Crst UMEntryThunkFreeList
+    AcquiredBefore LoaderHeap
```

Comment:
Ordering this with LoaderHeap should not be needed - you are not calling into LoaderHeap when the lock is taken.

Since your lock is a pretty simple leaf lock, you can just use CrstLeafLock for it and not add to this file at all.